### PR TITLE
GUACAMOLE-1809: Replace library used for IP address matching

### PR DESCRIPTION
**Pull request by Inperpetuammemoriam**
Newer versions of Spring Security lack support of Java 8.

### CODE REVIEW

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+        private Collection<String> trustedNetworks;
+
+        public MockConfigurationService() {
+            trustedNetworks = Collections.<String>emptyList();
+        }
+
+        public MockConfigurationService(String trustedNetworks) {
+            this.trustedNetworks = Arrays.asList(Pattern.compile(",\\s*").split(trustedNetworks));
+        }
```
Comment (mike-jumper) [inline extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java:55]: I'm OK with this as-is, but I suggest instead covering both these cases with just: `String... trustedNetworks`.

File: doc/licenses/spring-framework-3.0.5/README
Comment (mike-jumper) [inline doc/licenses/spring-framework-3.0.5/README:1]: This is actually used by the (rather unmaintained) JRadius library, which is pulled in only if building of extensions with LGPL-licensed dependencies is enabled via:

```
mvn -Plgpl-extensions package
```

Building with the above will result in the following error:

```
Processing runtime dependencies to produce LICENSE and NOTICE. Output will be within "/home/mjumper/apache/guacamole/guacamole-client/extensions/guacamole-auth-radius/target/licenses".
ERROR: License information missing for org.springframework:spring-context:jar:3.0.5.RELEASE
ERROR: License information missing for org.springframework:spring-aop:jar:3.0.5.RELEASE
ERROR: License information missing for org.springframework:spring-beans:jar:3.0.5.RELEASE
ERROR: License information missing for org.springframework:spring-core:jar:3.0.5.RELEASE
ERROR: License information missing for org.springframework:spring-expression:jar:3.0.5.RELEASE
ERROR: License information missing for org.springframework:spring-asm:jar:3.0.5.RELEASE
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
```

**Review by mike-jumper — CHANGES_REQUESTED** [review]

File: doc/licenses/spring-framework-3.0.5/README
Comment (Inperpetuammemoriam) [reply doc/licenses/spring-framework-3.0.5/README:1]: My bad! I didn't see that. I will revert this change.

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java (hunk 1 above)
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java:55]: Could you please be a little more specific. Do you mean that I should make use of just a single constructor?

File: extensions/guacamole-auth-json/src/main/java/org/apache/guacamole/auth/json/RequestValidationService.java [hunk 2]
```diff
@@ -77,23 +81,18 @@ public boolean isAuthenticationAllowed(HttpServletRequest request) {
...
-            if (matcher.matches(request)) {
+            if (new IPAddressString(network).contains(new IPAddressString(request.getRemoteAddr()))) {
                 logger.debug("Authentication request from \"{}\" is ALLOWED (matched subnet).", request.getRemoteAddr());
                 return true;
             }
 
         }
 
-        // Otherwise request is denied - no subnets matched
+        // Otherwise request is denied
```
[the remark this answers is not in the document]
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/main/java/org/apache/guacamole/auth/json/RequestValidationService.java:95]: Yes, it still is. I'm going to revert this.

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+    private RequestValidationService requestService;
+
+    private static HttpServletRequest mockHttpServletRequest(String remoteAddr) {
+
+        return new HttpServletRequest() {
+
+            public Object getAttribute(String name) {
+                return null;
+            }
```
[the remark this answers is not in the document]
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java:71]: Ok

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+            trustedNetworks = Collections.<String>emptyList();
+        }
+
+        public MockConfigurationService(String trustedNetworks) {
+            this.trustedNetworks = Arrays.asList(Pattern.compile(",\\s*").split(trustedNetworks));
+        }
+
+        public Collection<String> getTrustedNetworks() {
+            return trustedNetworks;
+        }
```
[the remark this answers is not in the document]
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java:59]: Ok

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+
+    }
+
+    private RequestValidationService requestService;
+
+    private static HttpServletRequest mockHttpServletRequest(String remoteAddr) {
```
[the remark this answers is not in the document]
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java:65]: Does the exception also include methods from an interface?

File: extensions/guacamole-auth-json/src/main/java/org/apache/guacamole/auth/json/RequestValidationService.java [hunk 6]
```diff
@@ -45,6 +45,10 @@ public class RequestValidationService {
     @Inject
     private ConfigurationService confService;
 
+    public RequestValidationService(ConfigurationService confService) {
```
[the remark this answers is not in the document]
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/main/java/org/apache/guacamole/auth/json/RequestValidationService.java:48]: Ok

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java (hunk 1 above)
Comment (mike-jumper) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java]: Yep. Reading through this, it occurred to me that what you have here is equivalent to:

```java
public MockConfigurationService(String... trustedNetworks) {
    this.trustedNetworks = Arrays.asList(trustedNetworks);
}
```

with the added benefit that you don't need to rely on string splitting.

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (mike-jumper) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java]: Yep. You don't need to worry about anything that is already JavaDoc'd and inherited from some other level.

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java (hunk 1 above)
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java]: > Yep. Reading through this, it occurred to me that what you have here is equivalent to:
> 
> ```java
> public MockConfigurationService(String... trustedNetworks) {
>     this.trustedNetworks = Arrays.asList(trustedNetworks);
> }
> ```
> 
> with the added benefit that you don't need to rely on string splitting.

Unfortunately, I cannot confirm that.

File: extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Inperpetuammemoriam) [reply extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java]: Ok. The documentation should then now be complete.

**Review by mike-jumper — APPROVED** [review]

**Pull request merged**